plugins/dell-dock2/fu-dell-dock2-wtpd.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2024 Dell Inc.
+ * All rights reserved.
+ *
+ * This software and associated documentation (if any) is furnished
+ * under a license and may only be used or copied in accordance
+ * with the terms of the license.
+ *
+ * This file is provided under a dual MIT/LGPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ * Dell Chooses the MIT license part of Dual MIT/LGPLv2 license agreement.
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-dell-dock2-common.h"
+
+struct _FuDellDock2Wtpd {
+	FuDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuDellDock2Wtpd, fu_dell_dock2_wtpd, FU_TYPE_DEVICE)
+
+static gchar *
+fu_dell_dock2_wtpd_ver_string(guint32 wtpd_version)
```

Comment:
all these device subclasses need to use fu_device_set_version_raw() with the uint value, and then they need to define a ->convert_version() that uses fu_version_from_uint32

Reply:
Fixed.